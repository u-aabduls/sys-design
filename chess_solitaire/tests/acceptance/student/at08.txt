-- ACCEPTANCE TEST #8: SUCCESS CASE: HARD

-- Setup some chess pieces for the game
setup_chess (N,1,2)
setup_chess (Q,2,3)
setup_chess (N,3,3)
setup_chess (P,3,4)
setup_chess (P,4,1)


-- Start the game
start_game


-- Show all possible moves for Knight@(1,2)
moves(1,2)

-- Show all possible moves for Queen@(2,3)
moves(2,3)

-- Show all possible moves for Knight@(3,3)
moves(3,3)

-- Show all possible moves for Pawn@(3,4)
moves(3,4)

-- Show all possible moves for Pawn@(4,1)
moves(4,1)


-- Queen capture Pawn
move_and_capture(2,3,3,4)

-- Show all possible moves for Queen@(3,4)
moves(3,4)

-- Queen capture Knight@(3,3)
move_and_capture(3,4,3,3)

-- Show all possible moves for Knight@(1,2)
moves(1,2)

-- Knight@(1,2) capture Queen
move_and_capture(1,2,3,3)

-- Show all possible moves for Knight@(3,3)
moves(3,3)

-- Knight capture Pawn
move_and_capture(3,3,4,1)


-- Game Over: You Win!