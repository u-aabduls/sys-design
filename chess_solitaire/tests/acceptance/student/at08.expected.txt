  # of chess pieces on board: 0
  Game being Setup...
  ....
  ....
  ....
  ....
->setup_chess(N,1,2)
  # of chess pieces on board: 1
  Game being Setup...
  .N..
  ....
  ....
  ....
->setup_chess(Q,2,3)
  # of chess pieces on board: 2
  Game being Setup...
  .N..
  ..Q.
  ....
  ....
->setup_chess(N,3,3)
  # of chess pieces on board: 3
  Game being Setup...
  .N..
  ..Q.
  ..N.
  ....
->setup_chess(P,3,4)
  # of chess pieces on board: 4
  Game being Setup...
  .N..
  ..Q.
  ..NP
  ....
->setup_chess(P,4,1)
  # of chess pieces on board: 5
  Game being Setup...
  .N..
  ..Q.
  ..NP
  P...
->start_game
  # of chess pieces on board: 5
  Game In Progress...
  .N..
  ..Q.
  ..NP
  P...
->moves(1,2)
  # of chess pieces on board: 5
  Game In Progress...
  .N..
  ...+
  +.+.
  ....
->moves(2,3)
  # of chess pieces on board: 5
  Game In Progress...
  .+++
  ++Q+
  .+++
  +.+.
->moves(3,3)
  # of chess pieces on board: 5
  Game In Progress...
  .+.+
  +...
  ..N.
  +...
->moves(3,4)
  # of chess pieces on board: 5
  Game In Progress...
  ....
  ..+.
  ...P
  ....
->moves(4,1)
  # of chess pieces on board: 5
  Game In Progress...
  ....
  ....
  .+..
  P...
->move_and_capture(2,3,3,4)
  # of chess pieces on board: 4
  Game In Progress...
  .N..
  ....
  ..NQ
  P...
->moves(3,4)
  # of chess pieces on board: 4
  Game In Progress...
  .+.+
  ..++
  +++Q
  ..++
->move_and_capture(3,4,3,3)
  # of chess pieces on board: 3
  Game In Progress...
  .N..
  ....
  ..Q.
  P...
->moves(1,2)
  # of chess pieces on board: 3
  Game In Progress...
  .N..
  ...+
  +.+.
  ....
->move_and_capture(1,2,3,3)
  # of chess pieces on board: 2
  Game In Progress...
  ....
  ....
  ..N.
  P...
->moves(3,3)
  # of chess pieces on board: 2
  Game In Progress...
  .+.+
  +...
  ..N.
  +...
->move_and_capture(3,3,4,1)
  # of chess pieces on board: 1
  Game Over: You Win!
  ....
  ....
  ....
  N...
